src/Symfony/Component/Form/Extension/Core/Type/TextType.php

```diff
@@ -28,6 +28,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         if ('' === $options['empty_data']) {
             $builder->addViewTransformer($this);
         }
+
+        $builder->addEventListener(
+            FormEvents::PRE_SUBMIT,
+            function (FormEvent $event) {
+                if (is_array($event->getData())) {
+                    $event->setData('');
+                    $event->getForm()->addError(
+                        new FormError('Invalid value.')
```

Comment:
I suggest you to add error this way:
```php
new FormError('This value should be of type {{ type }}.', null, ['{{ type }}' => 'string'])
```
Thus, it uses an already existing error message which is convenient for translations.
What do you think about it ?

Reply:
This solution is completely wrong. Read the comments above for details.